src/replication/replication_source_db_copy.cpp

```diff
@@ -357,7 +380,12 @@ namespace cubreplication
 
     er_log_debug_replication (ARG_FILE_LINE, "new_slave_copy connected");
 
-    /* TODO : next PR : extraction process : schema phase : start the client process */
+    /* extraction process : schema phase : start the client process */
+    error = locator_repl_extract_schema (&thread_ref, "dba", "");
+    if (error != NO_ERROR)
+      {
+	goto error;
+      }
 
     error = wait_receive_class_list (thread_ref);
```

Comment:
Do we need this call here? I think if we are here, the classes were received and no need to wait.

Reply:
The 'wait' here is not necessary. It may be enough to check the desired state.
However, I used 'wait' in the name of the method because it reuses the wait_state method mechanism; this includes the state check , but also the 'stop' flag (from the source_context module) and the interrupt flag from thread.